pallets/gear-program/src/migrations/allocations.rs

```diff
@@ -118,7 +118,7 @@ impl<T: Config> OnRuntimeUpgrade for MigrateAllocations<T> {
                 "incorrect count of elements old {} != new {}",
             );
             ensure!(
-                Pallet::<T>::current_storage_version() == MIGRATE_TO_VERSION,
+                Pallet::<T>::current_storage_version() == ALLOWED_CURRENT_STORAGE_VERSION,
```

Comment:
The `post_upgrade` hook is called after all migrations are applied, not just the current migration. Therefore, we need to check the final storage version and not the intermediate one.